packages/cli/create-strapi-app/src/create-strapi.ts

```diff
@@ -214,27 +214,84 @@ async function createApp(scope: Scope) {
 
 const installArguments = ['install'];
 
-const installArgumentsMap = {
+type VersionedArgumentsMap = {
+  [key: number]: string[]; // Maps version numbers to argument arrays
+};
+
+type VersionedEnvMap = {
+  [key: number]: Record<string, string>; // Maps version numbers to environment variables
+};
+
+// set command line options for specific package managers
+const installArgumentsMap: {
+  [key: string]: string[] | VersionedArgumentsMap;
+} = {
   npm: ['--legacy-peer-deps'],
-  yarn: ['--network-timeout 1000000'],
+  yarn: {
+    1: ['--network-timeout', '1000000'],
+    2: ['--network-timeout', '1000000'],
+    3: ['--network-timeout', '1000000'],
+  },
   pnpm: [],
 };
 
-function runInstall({ rootPath, packageManager }: Scope) {
+// set env vars for specific package managers
+const installEnvMap: {
+  [key: string]: Record<string, string> | VersionedEnvMap;
+} = {
+  yarn: {
+    4: { YARN_NETWORK_TIMEOUT: '1000000' },
+  },
+  npm: {},
+  pnpm: {},
+};
+
+async function getPackageManagerVersion(packageManager: string): Promise<number | null> {
+  try {
+    const { stdout } = await execa(packageManager, ['--version']);
+    const version = stdout.trim();
+    const majorVersion = parseInt(version.split('.')[0], 10);
```

Comment:
Should we use semver to parsde so we could have more precise check if needed ?

Reply:
yeah, I thought about that but worried it might complicate it even more than I did 😆
but I'll do that so we can (for example) make the assumption that yarn 5 will also need the env var with a range rather than adding in another major version

Reply:
👍